File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ImOpenEvent.java
package work.socialhub.kslack.entity.event;

/**
 * The im_open event is sent to all connections for a user when a direct message channel is opened by that user.
 * <p>
 * https://api.slack.com/events/im_open
 */
public class ImOpenEvent implements Event {

    public static final String TYPE_NAME = "im_open";

    private final String type = TYPE_NAME;
    private String channel;
    private String user;

    public String getType() {
        return this.type;
    }

    public String getChannel() {
        return this.channel;
    }

    public String getUser() {
        return this.user;
    }

    public void setChannel(String channel) {
        this.channel = channel;
    }

    public void setUser(String user) {
        this.user = user;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ImOpenEvent.kt
package work.socialhub.kslack.entity.event

/**
 * The im_open event is sent to all connections for a user when a direct message channel is opened by that user.
 *
 *
 * https://api.slack.com/events/im_open
 */
class ImOpenEvent : Event {
    override val type: String = TYPE_NAME
    var channel: String? = null
    var user: String? = null

    companion object {
        const val TYPE_NAME: String = "im_open"
    }
}